scripts/linkml_to_flat_synapse_jsonschema.py

```diff
@@ -0,0 +1,121 @@
+import subprocess
+import sys
+import json
+from pathlib import Path
+import jsonref
+
+def run_gen_json_schema(linkml_yaml, class_name, tmp_json):
+    cmd = f"gen-json-schema {linkml_yaml}"
+    if class_name:
+        cmd += f" --top-class {class_name}"
+    cmd += f" > {tmp_json}"
+    print(f"Running: {cmd}")
+    subprocess.run(cmd, shell=True, check=True)
+
+def flatten_json_schema(input_path, output_path):
+    """Flatten/dereference $ref in a JSON Schema file using Python (jsonref)."""
+    with open(input_path, 'r') as f:
+        schema = json.load(f)
+    # Dereference $ref
+    deref_schema = jsonref.JsonRef.replace_refs(schema)
+    # Convert to plain dict to ensure JSON serializable
+    def convert_to_plain_dict(obj):
+        if isinstance(obj, dict):
+            return {k: convert_to_plain_dict(v) for k, v in obj.items()}
+        elif isinstance(obj, list):
+            return [convert_to_plain_dict(item) for item in obj]
+        else:
+            return obj
```

Comment:
I'm personally not a huge fan of nested functions because you can't test them.  I would pull this out and make it a "hidden" and/or helper function with a `_` in front of it.